bioptim/limits/phase_transition.py

```diff
@@ -228,22 +228,29 @@ def continuous(transition, all_pn):
             -------
             The difference between the state after and before
             """
-
             nlp_pre, nlp_post = all_pn[0].nlp, all_pn[1].nlp
 
             if transition.states_pre_idx is None:
-                transition.states_pre_idx = list(range(nlp_pre.states.cx.shape[0]))
+                transition.states_pre_idx = list(range(nlp_pre.states.cx_end.shape[0]))
                 transition.states_post_idx = list(range(nlp_post.states.cx.shape[0]))
 
-            states_pre = vertcat(*[nlp_pre.states.cx_end[i] for i in transition.states_pre_idx])
-            states_post = vertcat(*[nlp_post.states.cx[i] for i in transition.states_post_idx])
+            states_pre = []
+            states_post = []
+            for idx in range(len(transition.states_pre_idx)):
+                states_pre = vertcat(states_pre, nlp_pre.states.cx_end[transition.states_pre_idx[idx]])
+                states_post = vertcat(states_post, nlp_post.states.cx[transition.states_post_idx[idx]])
 
             if states_pre.shape != states_post.shape:
                 raise RuntimeError(
-                    f"Continuity can't be established since the number of x to be matched is {states_pre.shape} in the pre-transition phase and {states_post.shape} post-transition phase.")
+                    f"Continuity can't be established since the number of x to be matched is {states_pre.shape} in the "
+                    f"pre-transition phase and {states_post.shape} post-transition phase."
+                )
 
             continuity = states_pre - states_post
-            return continuity
+
+            name = f"PHASE_TRANSITION_{nlp_pre.phase_idx}_{nlp_post.phase_idx}"
+            func = Function(name, [states_pre, states_post], [continuity])(states_pre, states_post)
```

Comment:
I feel this is dangerous when use_sx is True.
Have you tested with both?
Also in order to be consistent with the rest of the code, Function is never manually called, but done via `biorbd.to_casadi_func(name, function, *mx)`. In your case, that would mean:
`biorbd.to_casadi_func(name, continuity, [states_pre, states_post], transition.expand)`
I am not sure "transition.expand", if not, just send True

Reply:
I can not use to_casadi_func since it creates free variables :
"Function with free symbols cannot be expanded. List of free variables in your Function: q_Seg1_RotX_1,q_Seg2_RotX_1,qdot_0_1,qdot_1_1,q_Seg1_RotX_0,q_Seg2_RotX_0,qdot_2_0,qdot_3_0"